src/onedrive/onedrive-signon/onedrivesignonsyncadaptor.cpp

```diff
@@ -0,0 +1,305 @@
+/****************************************************************************
+ **
+ ** Copyright (C) 2015 Jolla Ltd.
+ ** Contact: Chris Adams <chris.adams@jollamobile.com>
+ **
+ ** This program/library is free software; you can redistribute it and/or
+ ** modify it under the terms of the GNU Lesser General Public License
+ ** version 2.1 as published by the Free Software Foundation.
+ **
+ ** This program/library is distributed in the hope that it will be useful,
+ ** but WITHOUT ANY WARRANTY; without even the implied warranty of
+ ** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ ** Lesser General Public License for more details.
+ **
+ ** You should have received a copy of the GNU Lesser General Public
+ ** License along with this program/library; if not, write to the Free
+ ** Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ ** 02110-1301 USA
+ **
+ ****************************************************************************/
+
+#include "onedrivesignonsyncadaptor.h"
+#include "trace.h"
+
+#include <QtCore/QPair>
+#include <QtCore/QJsonArray>
+#include <QtCore/QJsonValue>
+#include <QtCore/QJsonObject>
+#include <QtCore/QJsonDocument>
+#include <QtCore/QUrlQuery>
+#include <QtCore/QTimer>
+
+OneDriveSignonSyncAdaptor::OneDriveSignonSyncAdaptor(QObject *parent)
+    : OneDriveDataTypeSyncAdaptor(SocialNetworkSyncAdaptor::Signon, parent)
+{
+    setInitialActive(true);
+}
+
+OneDriveSignonSyncAdaptor::~OneDriveSignonSyncAdaptor()
+{
+}
+
+QString OneDriveSignonSyncAdaptor::syncServiceName() const
+{
+    return QStringLiteral("onedrive-sync"); // TODO: change name of service to onedrive-signon!
```

Comment:
depending on the a&sso .service id we use for this, it should be onedrive-signon